ProjFS.Mac/PrjFSKext/VnodeCache.cpp

```diff
@@ -177,10 +183,26 @@ KEXT_STATIC_INLINE void InvalidateCache_ExclusiveLocked()
     memset(s_entries, 0, s_entriesCapacity * sizeof(VnodeCacheEntry));
 }
 
+KEXT_STATIC_INLINE uint32_t ComputePow2CacheCapacity(int expectedVnodeCount)
+{
+    uint32_t idealCacheSize = expectedVnodeCount * 2;
+    size_t index = 0;
+    uint32_t capacity;
+    
+    do
+    {
+        capacity = AllowedPow2CacheCapacities[index];
+        ++index;
+    }
+    while (capacity < idealCacheSize && index < Array_Size(AllowedPow2CacheCapacities));
+    
+    return capacity;
+}
```

Comment:
Note that you could use the base-2 logarithm function we use for computing time buckets in the perf tracer to fairly easily get the next-highest power of 2 if you prefer. Also, in the worst case you could end up with just over 25% load factor with the current algorithm (if `expectedVnodeCount` is only slightly larger than a power of 2). With a good hash function, aiming for 50-80% is probably more appropriate. So perhaps get the next largest power of 2 and only double the capacity if `expectedVnodeCount` is more than 80% of that power of 2?

Reply:
I've switched to using `<< 1` to calculate the powers of two.  The question of when to move up to the next power of 2 will be investigated as part of #974